internal/api/verify.go

```diff
@@ -494,6 +495,14 @@ func (a *API) verifyUserAndToken(ctx context.Context, conn *storage.Connection,
 			user, err = models.FindUserByPhoneChangeAndAudience(conn, params.Phone, aud)
 		case smsVerification:
 			user, err = models.FindUserByPhoneAndAudience(conn, params.Phone, aud)
+			smsProvider, _ := sms_provider.GetSmsProvider(*config)
+			if config.Sms.Provider == "twilio" && config.Sms.Twilio.VerifyEnabled {
+				if err := smsProvider.VerifyOTP(params.Phone, params.Token); err != nil {
+					return nil, expiredTokenError("Token has expired or is invalid").WithInternalError(err)
+				}
+				return user, nil
+			}
+
```

Comment:
Why is this necessary? Shouldn't it be exactly the same as the existing providers?

Reply:
Yup, you are right this section verifies the user and should be the same as existing providers. We still need to  verify the token with Twilio's Verification Service and that would probably make more sense if done before the `isOtpValid` check.

Let me know if there's anything I'm missing.